cpp/tests/reductions/ewm_tests.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2021-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "scan_tests.hpp"
+
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+
+#include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/reduction.hpp>
+
+using aggregation = cudf::aggregation;
+using cudf::null_policy;
+using cudf::scan_type;
+using namespace cudf::test::iterators;
+
+template <typename T>
+struct TypedEwmScanTest : BaseScanTest<T> {
+  inline void test_ungrouped_ewma_scan(cudf::column_view const& input,
+                                       cudf::column_view const& expect_vals,
+                                       cudf::scan_aggregation const& agg,
+                                       null_policy null_handling)
+  {
+    auto col_out = cudf::scan(input, agg, scan_type::INCLUSIVE, null_handling);
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expect_vals, col_out->view());
+  }
+};
+
+TYPED_TEST_SUITE(TypedEwmScanTest, cudf::test::FloatingPointTypes);
+
+TYPED_TEST(TypedEwmScanTest, Ewm)
+{
+  auto const v = make_vector<TypeParam>({1.0, 2.0, 3.0, 4.0, 5.0});
+  auto col     = this->make_column(v);
+
+  auto const expected_ewma_vals_adjust = cudf::test::fixed_width_column_wrapper<TypeParam>{
+    {1.0, 1.75, 2.61538461538461497469, 3.54999999999999982236, 4.52066115702479365268}};
+
+  auto const expected_ewma_vals_noadjust =
+    cudf::test::fixed_width_column_wrapper<TypeParam>{{1.0,
+                                                       1.66666666666666651864,
+                                                       2.55555555555555535818,
+                                                       3.51851851851851815667,
+                                                       4.50617283950617242283}};
```

Comment:
These are a lot of digits of precision. Maybe we should consider rounding off some digits and comparing that instead or something? I'm not sure.